src/JustEat.StatsD/V2/StatsDPublisherV2.cs

```diff
@@ -0,0 +1,170 @@
+using System;
+using System.Threading;
+using JustEat.StatsD.EndpointLookups;
+
+namespace JustEat.StatsD.V2
+{
+    public sealed class StatsDPublisherV2 : IStatsDPublisher
```

Comment:
I feel that it would be better to have the original publisher's implementation details be able to handle `IStatsDPublisher` and `IStatsDPublisherV2` so we don't need all these "V2" alternate versions which double the amount of code to maintain.